### PR TITLE
Fix : fix bug in modulebuilder and fix php8.0 warnings in modulebuilder

### PR DESCRIPTION
**Pull request by Hystepik**
fix bug while class and sql regeneration in modulebuilder
fix php8.0 warnings in modulebuilder

### CODE REVIEW

File: htdocs/core/lib/modulebuilder.lib.php [hunk 1]
```diff
@@ -125,46 +125,46 @@ function rebuildObjectClass($destdir, $module, $objectname, $newmask, $readdir =
 				$texttoinsert .= " 'position'=>".($val['position'] !== '' ? $val['position'] : 50).",";
 				$texttoinsert .= " 'notnull'=>".(empty($val['notnull']) ? 0 : $val['notnull']).",";
 				$texttoinsert .= " 'visible'=>".($val['visible'] !== '' ? $val['visible'] : -1).",";
-				if ($val['noteditable']) {
+				if (!empty($val['noteditable'])) {
 					$texttoinsert .= " 'noteditable'=>'".$val['noteditable']."',";
 				}
-				if ($val['default'] || $val['default'] === '0') {
+				if (!empty($val['default']) || (!empty($val['default']) && $val['default'] === '0')) {
```
[the remark this answers is not in the document]
Comment (eldy) [reply htdocs/core/lib/modulebuilder.lib.php]: @Hystepik still strange
if !empty then another test on !empty  after a OR
I think it should be   isset($val['default'])  for the second one.
